examples/java/src/main/java/examples/ExceptionHandlerServerMiddleware.java

```
@@ -0,0 +1,53 @@
+package examples;
+
+import com.workiva.frugal.middleware.InvocationHandler;
+import com.workiva.frugal.middleware.ServiceMiddleware;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import org.apache.thrift.TApplicationException;
+import org.apache.thrift.TException;
+import org.apache.thrift.transport.TTransportException;
+import v1.music.PurchasingError;
+
+public class ExceptionHandlerServerMiddleware implements ServiceMiddleware {
+
+    @Override
+    public <T> InvocationHandler<T> apply(T next) {
+        return new InvocationHandler<T>(next) {
+            @Override
+            public Object invoke(Method method, Object receiver, Object[] args) throws Throwable {
+                System.out.printf("==== CALLING %s.%s ====\n", method.getDeclaringClass().getName(), method.getName());
+
+                // Default error code is UNKNOWN
+                int errorCode = TApplicationException.UNKNOWN;
+                try {
+                    try {
+                        Object result = method.invoke(receiver, args);
+                        return result;
+                    } catch (InvocationTargetException e) {
+                        // Throw underlying exception
+                        throw e.getTargetException();
+                    }
+                } catch (TApplicationException e) {
+                    // See https://github.com/Workiva/frugal/blob/master/lib/java/src/main/java/com/workiva/frugal/exception/TApplicationExceptionType.java
+                    // and https://github.com/Workiva/messaging-sdk/blob/master/lib/java/sdk/src/main/java/com/workiva/messaging_sdk/exception/TApplicationExceptionType.java
+                    errorCode = e.getType();
```

It seems a poor example to swallow the exception.  Perhaps rethrow and put the `errorCode` logging in the `finally`?

To be sure, server middleware won't see INTERNAL_ERROR for uncaught `RuntimeException` (like `NullPointerException`) because that translation doesn't happen until `FBaseProcessor`, so you have to catch those (or `Throwable`) explicitly.  I think updating `FStoreHandler` to periodically throw `NullPointerException` would show that.

I thought we had done work to also handle `Error` (and "sneaky throws"), but it looks like that only happened for NATS subscription ([code](https://github.com/Workiva/frugal/blob/3cf3c22ee15636de2cf2790eb5fd968e637429f6/lib/java/src/main/java/com/workiva/frugal/transport/FNatsSubscriberTransport.java#L120-L121)).  For RPC, I think NATS drops the message (client timeout) and HTTP would return 500 (rather than TApplicationException.INTERNAL_ERROR), so I think `FHttpTransport` will convert that to a vague `TTransportException`, which seems like it will conflate "client<->server network error" and "internal service error".